include/yaml_cpp_catkin/yaml_tools.hpp

```diff
@@ -8,35 +8,68 @@
 
 #pragma once
 
-#include <yaml-cpp/yaml.h>
+#include <iomanip>
+#include <yaml_cpp_catkin/yaml_eigen.h>
 
 namespace YAML {
 
-  /**
-   * @brief helper function to safely read a yaml parameter
-   * 
-   * @tparam YamlType 
-   * @param node 
-   * @param name 
-   * @return YamlType 
-   */
-  template <typename YamlType>
-  static YamlType ReadParameter(const YAML::Node& node, const std::string& name) {
-    try { return node[name.c_str()].as<YamlType>(); }
-    catch (...) { throw std::runtime_error("Error reading the yaml parameter [" + name + "]"); }
-  }
+/**
+ * @brief helper function to safely read a yaml parameter
+ *
+ * @tparam YamlType
+ * @param node
+ * @param name
+ * @return YamlType
+ */
+template<typename YamlType>
+static YamlType ReadParameter(const YAML::Node& node, const std::string& name) {
+	try {
+		return node[name.c_str()].as<YamlType>();
+	} catch (...) {
+		throw std::runtime_error(
+				"Error reading the yaml parameter [" + name + "]");
+	}
+}
 
-  /**
-   * @brief helper function to safely read a yaml parameter
-   * 
-   * @tparam YamlType 
-   * @param node 
-   * @param name 
-   * @param parameter 
-   */
-  template <typename YamlType>
-  static void ReadParameter(const YAML::Node& node, const std::string& name, YamlType& parameter) {
-    parameter = ReadParameter<YamlType>(node, name);
-  }
+/**
+ * @brief helper function to safely read a yaml parameter
+ *
+ * @tparam YamlType
+ * @param node
+ * @param name
+ * @param parameter
+ */
+template<typename YamlType>
+static void ReadParameter(const YAML::Node& node, const std::string& name,
+		YamlType& parameter) {
+	parameter = ReadParameter<YamlType>(node, name);
+}
+
+/**
+ * @brief helper function to safely read a yaml parameter
+ *
+ * @param node
+ * @param name
+ * @tparam YamlType
+ * @param optional
+ * @return YamlType
+ */
+template<typename YamlType>
+void readParameter(const YAML::Node& node, const std::string& name,
+		YamlType& parameter, bool optional = false) {
+	try {
```

Comment:
This could be simplified:
```
if (optional && !node[name.c_str()])
{
   return;
}
try{
    parameter = node[name.c_str()].as<YamlType>();
}catch(...){
    throw std::runtime_error(name);
}
```

Reply:
Good one !! Thanks !!

Reply:
Can you fix the code and push to the same branch when you have time?